packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/project/controllers/ProjectRestController.java

```diff
@@ -0,0 +1,120 @@
+import org.eclipse.sirius.web.application.project.services.api.IProjectApplicationService;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * REST Controller for the Project Endpoint.
+ *
+ * @author arichard
+ */
+@RestController
+@RequestMapping("/api/rest/projects")
+public class ProjectRestController {
+
+    private final IProjectApplicationService projectApplicationService;
+
+    public ProjectRestController(IProjectApplicationService projectApplicationService) {
+        this.projectApplicationService = Objects.requireNonNull(projectApplicationService);
+    }
+
+    @GetMapping
+    public ResponseEntity<List<RestProject>> getProjects() {
+        var restProjects = this.projectApplicationService.findAll(PageRequest.of(0, 20))
+            .map(project -> new RestProject(project.id(), null, List.of(), null, null, project.name(), List.of()))
+            .toList();
+
+        return new ResponseEntity<>(restProjects, HttpStatus.OK);
+    }
+
+    @GetMapping(path = "/{projectId}")
+    public ResponseEntity<RestProject> getProjectById(@PathVariable UUID projectId) {
+        var restProject = this.projectApplicationService.findById(projectId)
+            .map(project -> new RestProject(project.id(), null, List.of(), null, null, project.name(), List.of()));
+
+        if (restProject.isPresent()) {
+            return new ResponseEntity<>(restProject.get(), HttpStatus.OK);
+        }
+
+        return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+    }
+
+    @PostMapping
+    public ResponseEntity<RestProject> createProject(@RequestParam String name, @RequestParam Optional<String> description) {
+        var createProjectInput = new CreateProjectInput(UUID.randomUUID(), name, List.of());
+        var newProjectPayload = this.projectApplicationService.createProject(createProjectInput);
+
+        if (newProjectPayload instanceof CreateProjectSuccessPayload createProjectSuccessPayload) {
+            var projectDTO = createProjectSuccessPayload.project();
+            var restProject = new RestProject(projectDTO.id(), null, List.of(), null, null, projectDTO.name(), List.of());
+            return new ResponseEntity<>(restProject, HttpStatus.CREATED);
+        }
+        // The specification does not handle other HttpStatus than HttpStatus.CREATED for this endpoint
+        return null;
```

Comment:
I should test that but if you know, what happens in this case? Do we have at least a proper HTTP response?

Reply:
Yes, but a `200 OK` one. I don't know If we should return another kind of HTTP response, even if the specification does not handle the case.

Reply:
If it fallback on a default value that's ok for me. The specification should probably indicate what happens if when can't perform the action, the obvious one would be BAD_REQUEST but who knows.

:+1: